src/qservkafka/models/kafka.py

```diff
@@ -0,0 +1,315 @@
+"""Models for Kafka messages."""
+
+from __future__ import annotations
+
+from datetime import UTC, datetime
+from enum import StrEnum
+from typing import Annotated
+
+from pydantic import BaseModel, ConfigDict, Field, HttpUrl
+from safir.pydantic import SecondsTimedelta, UtcDatetime
+from vo_models.uws.types import ExecutionPhase
+
+__all__ = [
+    "JobError",
+    "JobMetadata",
+    "JobQuery",
+    "JobResult",
+    "JobResultEnvelope",
+    "JobResultFormat",
+    "JobResultType",
+    "JobRun",
+    "JobStatus",
+]
+
+
+class JobResultEnvelope(BaseModel):
+    """VOTable envelope for job results."""
+
+    model_config = ConfigDict(validate_by_name=True)
+
+    header: Annotated[
+        str,
+        Field(
+            title="Result XML header",
+            description="VOTable XML header for the results",
+        ),
+    ]
+
+    footer: Annotated[
+        str,
+        Field(
+            title="Result XML footer",
+            description="VOTable XML footer for the results",
+        ),
+    ]
+
+    base_url: Annotated[
+        str | None,
+        Field(
+            title="Base URL for access_url",
+            description="Base URL for rewriting access_url column values",
+            validation_alias="baseUrl",
+        ),
+    ] = None
+
+
+class JobResultType(StrEnum):
+    """Possible types for the output format of results."""
+
+    votable = "votable"
+
+
+class JobResultFormat(BaseModel):
+    """Result format for job results."""
+
+    model_config = ConfigDict(validate_by_name=True)
+
+    type: Annotated[
+        JobResultType,
+        Field(
+            title="Output format for result",
+            description="Format in which to write the output",
+        ),
+    ]
+
```

Comment:
I agree with your point, it may be good to add the serialization here (even though I'm not sure if we do anything with it in TAP):
```

serialization: Annotated[
    str | None,
    Field(
        title="Serialization type",
        description="How the result should be serialized (e.g., binary2, tabledata)",
    ),
] = None
```